core_backend/app/question_answer/routers.py

```diff
@@ -187,54 +236,68 @@ async def get_semantic_matches(
 
 @router.post("/response-feedback")
 async def feedback(
-    feedback: ResponseFeedbackBase, asession: AsyncSession = Depends(get_async_session)
+    feedback: ResponseFeedbackBase,
+    asession: AsyncSession = Depends(get_async_session),
+    auth_token: str = Depends(auth_bearer_token),
 ) -> JSONResponse:
     """
     Feedback endpoint used to capture user feedback on the results returned.
     <BR>
     <B>Note</B>: If you wish to only provide `feedback_text`, don't include
     `feedback_sentiment` in the payload.
     """
+    user = await get_user_by_token(auth_token, asession)
+    if user is None:
+        return JSONResponse(
+            status_code=400,
+            content={"message": "No user associated with the provided token."},
+        )
```

Comment:
Hmmm... i agree that it is not needed since feedback secret key is associated with a unique query. So if you leak your feedback secret key, someone else can provide feedback for your query.

I can't decide if that's a feature or a bug but leaning towards it's a good thing.

There is something nice about passing on a feedback key to some other service without sharing the bearer token to allow them to forward feedback.

Reply:
Hmm but to call the feedback endpoints you need the API key anyway so it's not that another service could add feedback without having some user's API key.

I think I'll leave it in for consistency with the other endpoints for now. I think it actually makes sense to check if the user exists during auth, so this will become outdated in the last PR anyway.